pkg/controller/master-controller-manager/kubeone/controller.go

```diff
@@ -122,19 +200,121 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
-	err := r.reconcile(ctx, externalCluster)
+	if externalCluster.DeletionTimestamp != nil {
+		log.Debug("Deleting KubeOne Namespace")
+		ns := &corev1.Namespace{}
+		name := types.NamespacedName{Name: kubernetesprovider.GetKubeOneNameSpaceName(externalCluster.Name)}
```

Comment:
Please settle on either "Namespace" or "NameSpace", don't mix the two. Also, please settle on "Namespace" ;-)